main.cpp

```diff
@@ -4296,16 +4298,17 @@ bool load_guts(picoboot::connection con, iostream_memory_access &file_access) {
                     raw_access.write_vector(base, file_buf);
                     base += this_batch;
                 }
-                bar.progress(base - mem_range.from, mem_range.to - mem_range.from);
+                bar.progress(base - mem_range.from + loaded_bytes, total_bytes);
             }
+            loaded_bytes += mem_range.to - mem_range.from;
         }
     }
-    for (auto mem_range : ranges) {
-        enum memory_type type = get_memory_type(mem_range.from, model);
-        if (settings.load.verify) {
+    if (settings.load.verify) {
+        progress_bar bar("Verifying: ");
+        size_t verified_bytes = 0;
+        for (auto mem_range : ranges) {
             bool ok = true;
             {
```

Comment:
And I guess a similar thing with the "extra scope" here?

Reply:
There's no extra scope here, the scope of the `if` statement is exactly what we need for the progress bar.

Reply:
Doesn't the `{` under the `bool ok = true;` line introduce an additional scope?

Reply:
Oh yeah, that's an extra scope.  I didn't want to mess with it because it would make my diff a lot bigger to remove that extra block and unindent all the code inside it.

Reply:
Fair enough :slightly_smiling_face: